src/components/search/search.tsx

```diff
@@ -1,44 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import classes from './search.module.scss';
 
-interface ISearch {
-  textSearch: string;
-}
+const Search = () => {
+  const [textSearch, setSearch] = useState(localStorage.getItem('LocalStorageSearch') || '');
 
-class Search extends React.Component<object, ISearch> {
-  constructor(props: object) {
-    super(props);
-    this.state = {
-      textSearch: localStorage.getItem('LocalStorageSearch') || '',
-    };
-    this.changeText = this.changeText.bind(this);
-  }
+  const changeText = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setSearch(e.target.value);
+  };
 
-  changeText(e: React.ChangeEvent<HTMLInputElement>): void {
-    this.setState({
-      textSearch: e.target.value,
-    });
-  }
-  componentWillUnmount(): void {
-    localStorage.setItem('LocalStorageSearch', `${this.state.textSearch}`);
-  }
+  useEffect(() => {
+    localStorage.setItem('LocalStorageSearch', `${textSearch}`);
```

Comment:
u can move it to the changeText and there will be no need to call useEffect every time u change value